### PR TITLE
Fix bugs with constraints when using local smoothing

### PR DESCRIPTION
**Pull request by lpsaavedra**
### Description

The Taylor-Couette transient case was not converging when using local smoothing. 

### Solution

Several bugs were identified and fixed:
- Pressure constraint was not correctly set for local smoothing.
- A bug in the periodic boundary conditions was introduced since an additional boundary type `periodic_neighbor` was introduced in [](https://github.com/chaos-polymtl/lethe/pull/1330).
- A timer that was throwing an error in debug mode.

### Testing

A minimal Taylor-Couette test with periodic boundary conditions was added to ensure that this does not happen again when using `lsmg` and `gcmg` as preconditioners for the matrix-free application. 

### Documentation

It does not change anything regarding the documentation. 

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [ ] All in-code documentation related to this PR is up to date (Doxygen format)
- [ ] Copyright headers are present and up to date
- [ ] Lethe documentation is up to date
- [ ] Fix has unit test(s) (preferred) or application test(s), and restart files are in the generator folder
- [ ] The branch is rebased onto master
- [ ] Changelog (CHANGELOG.md) is up to date
- [ ] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [ ] Labels are applied
- [ ] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [ ] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [ ] If the fix is temporary, an issue is opened
- [ ] The PR description is cleaned and ready for merge

### CODE REVIEW

File: source/solvers/fluid_dynamics_matrix_free.cc [hunk 1]
```diff
@@ -1133,6 +1143,11 @@ MFNavierStokesPreconditionGMGBase<dim>::reinit(
                       .periodic_direction.at(id),
                     level_constraint);
                 }
+              else if (type ==
+                       BoundaryConditions::BoundaryType::periodic_neighbor)
+                {
+                  /*already taken into account in the previous else if*/
+                }
```
Comment (blaisb) [inline source/solvers/fluid_dynamics_matrix_free.cc:1158]: might be better to migrate to if statements here instead of if else if else if else since I feel like these problems are bout to arise.
Maybe at least you could have the "else statement" throw an error instead something like this boundary condition is not defined. Otherwise i am afraid these errors can happen again in the future

Comment (lpsaavedra) [reply source/solvers/fluid_dynamics_matrix_free.cc:1158]: I agree, I have updated it accordingly. Now we check for each of the cases and throw in case a non-supported BC is specified.

**Comment by blaisb** [comment]
Will merge the other PR and rebase this one after when I wake up tmrow
 :)
